scripts/cbmc-viewer/errors.py

```diff
@@ -0,0 +1,139 @@
+"""Format report of property violations found by cbmc."""
+
+import re
+
+ERROR = 0
+WARNING = 1
+PROP = "MISSINGPROP"
+LOOP = "MISSINGLOOP"
+
+
+def assemble_report(result, properties):
+    """Map property violations in result to source line of the violation."""
+    report = {}
+    report[PROP] = {}
+    report[PROP][ERROR] = []
+    report[PROP][WARNING] = []
+    report[LOOP] = {}
+    report[LOOP][ERROR] = []
+    report[LOOP][WARNING] = []
+    for name in result.failures():
+        kind = ERROR if result.error.get(name) else WARNING
+        prop = properties.lookup(name)
+        if not prop:
+            print("Warning: Can't find property {}".format(name))
+            if re.match(r'.*\.unwind\.[0-9]+$', name):
+                report[LOOP][kind].append(name)
+            else:
+                report[PROP][kind].append(name)
+            continue
+        func = prop["function"]
+        src = prop["file"]
+        line = int(prop["line"])
+        desc = prop["description"]
+
+        if not report.get(src):
+            report[src] = {}
+        if not report[src].get(func):
+            report[src][func] = {}
+        if not report[src][func].get(line):
+            report[src][func][line] = {}
+        if not report[src][func][line].get(ERROR):
+            report[src][func][line][ERROR] = []
+        if not report[src][func][line].get(WARNING):
+            report[src][func][line][WARNING] = []
+        report[src][func][line][kind].append((name, desc))
+    report[PROP][ERROR].sort()
+    report[PROP][WARNING].sort()
+    report[LOOP][ERROR].sort()
+    report[LOOP][WARNING].sort()
+    return report
+
+
+def format_report(report, markup, loops=None, traces=None):
+    """Format report on property violations."""
+
+    # pylint: disable=too-many-branches
+
+    html = ['<div style="font-family:courier">']
```

Comment:
Inserting this comment at a fairly random location, but it's a general comment regarding the HTML output - rather than specifying style info in hard coded ways such as this local `font-family` tag - could you do it all on a style sheet (even if the stylesheet is embedded in the generated HTML) - that would allow other tools to potentially adjust the style to their own local conventions in a much more flexible way - possibly even allowing various JavaScript DOM hackery to take place by other clients (I'm not a web programmer, so excuse my ignorance of exactly what is doable there).

Reply:
This is added to the TODO.md list of things to do.